tests/php/Forms/DateField_DisabledTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace SilverStripe\Forms\Tests;
+
+use IntlDateFormatter;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Forms\DateField_Disabled;
+use SilverStripe\Forms\RequiredFields;
+use SilverStripe\i18n\i18n;
+use SilverStripe\ORM\FieldType\DBDatetime;
+
+/**
+ * @skipUpgrade
+ */
+class DateField_DisabledTest extends SapphireTest
+{
+    protected function setUp()
+    {
+        parent::setUp();
+        i18n::set_locale('en_NZ');
+        DBDatetime::set_mock_now('2011-02-01 8:34:00');
+    }
+
+    public function testField()
+    {
+        // Today date with normal local
+        $actual = DateField_Disabled::create('Test')
+            ->setValue('2011-02-01')
+            ->Field();
+        $expected = '<span class="readonly" id="Test">1/02/2011 (today)</span>';
+        $this->assertEquals($expected, $actual);
+
+        // Test today's date with time
+        $actual = DateField_Disabled::create('Test')
+            ->setValue('2011-02-01 10:34:00')
+            ->Field();
+        $this->assertEquals($expected, $actual);
+
+        // Test yesterday
+        $actual = DateField_Disabled::create('Test')
+            ->setValue('2011-01-27')
+            ->Field();
+        $expected = '<span class="readonly" id="Test">27/01/2011, 5 days ago</span>';
+        $this->assertEquals($expected, $actual);
+
+        // Test tomorrow
+        $actual = DateField_Disabled::create('Test')
+            ->setValue('2011-02-06')
+            ->Field();
+        $expected = '<span class="readonly" id="Test">6/02/2011, in 5 days</span>';
+        $this->assertEquals($expected, $actual);
+
+        // Test different local
+        $actual = DateField_Disabled::create('Test')
+            ->setValue('2011-02-06')
+            ->setHTML5(false)
+            ->setLocale('de_DE')
+            ->Field();
+        $expected = '<span class="readonly" id="Test">06.02.2011, in 5 days</span>';
+        $this->assertEquals($expected, $actual);
+
+        // Test none value
+        $actual = DateField_Disabled::create('Test')->Field();
+        $expected = '<span class="readonly" id="Test"><i>(not set)</i></span>';
+        $this->assertEquals($expected, $actual);
+
+        $actual = DateField_Disabled::create('Test')->setValue('This is not a date')->Field();
+        $this->assertEquals($expected, $actual);
```

Comment:
This is pretty minor but these look like they should be separate unit tests to me

Reply:
You mean you would put the two none value test in their own _test_ function?

Reply:
Yep :-) to me a unit test should test a single aspect of the code

Reply:
Fair point. Just did it.

Reply:
Cool - there's still five separate scenarios in `testField` that could be separated. They're currently annotated with comments

Reply:
You're a tough customer Robbie :wink: 

I've split up the test further with an individual test for:
* the local
* today's date
* different day from today